@here/harp-omv-datasource/test/MapViewPickingTest.ts

```diff
@@ -0,0 +1,326 @@
+                    (textElem.objInfos![0] as any).name,
+                    GEOJSON_DATA.features[2].properties.name
+                );
+            });
+
+            it("Decoded tile contains line pick data pointer", () => {
+                lineGeometry = decodeTile.geometries!.find(
+                    geometry => geometry.type === GeometryType.SolidLine
+                );
+
+                assert.notEqual(lineGeometry, undefined);
+                assert.notEqual(lineGeometry!.groups, undefined);
+                assert.equal(lineGeometry!.groups.length, 1);
+            });
+
+            it("TilInfo contains line pick data", () => {
+                assert.notEqual(tileInfo.lineGroup, undefined);
+
+                assert.notEqual(tileInfo.lineGroup.positionIndex, undefined);
+                assert.equal(tileInfo.lineGroup.positionIndex.length, 1);
+
+                assert.notEqual(tileInfo.lineGroup.userData, undefined);
+                assert.equal(tileInfo.lineGroup.userData.length, 1);
+
+                const index = tileInfo.lineGroup.positionIndex[0] as number;
+                assert.equal(lineGeometry!.groups[0].start, index);
+
+                const userData = tileInfo.lineGroup.userData[index] as any;
+                assert.notEqual(userData, undefined);
+                assert.equal(userData.name, GEOJSON_DATA.features[1].properties!.name);
+            });
+
+            it("decodedTile contains polygon objInfos", () => {
+                const polygonGeometry = decodeTile.geometries!.find(
+                    geometry => geometry.type === GeometryType.Polygon
+                ) as Geometry;
+
+                assert.notEqual(polygonGeometry, undefined, "polygon geometry missing");
+                assert.notEqual(
+                    polygonGeometry.groups,
+                    undefined,
+                    "polygon geometry groups missing"
+                );
+                assert.equal(polygonGeometry.groups.length, 1);
+                assert.notEqual(polygonGeometry.objInfos, undefined, "objInfos missing");
+                assert.equal(polygonGeometry.objInfos!.length, 1);
+
+                const objInfo = polygonGeometry.objInfos![0] as any;
+                assert.equal(objInfo.name, GEOJSON_DATA.features[0].properties.name);
+            });
+        });
+    });
+
+    // emulate a real pick in browser
+    it("Pick polygon", async () => {
+        const POLYGON_DATA = GEOJSON_DATA.features[0];
+        const POINT = GEOJSON_DATA.features[3];
+        const pointCoordinates = ((POINT.geometry as any).coordinates as any) as number[];
+
+        const point = new GeoCoordinates(pointCoordinates[1], pointCoordinates[0]);
+        await displayLocation(mapView, point);
+
+        const screenPointLocation = mapView.getScreenPosition(point) as THREE.Vector2;
+        assert.notEqual(screenPointLocation, undefined);
+
+        mapView.worldRootObject.children.forEach(object => {
```

Comment:
This is redundant. You can do `updateMatrixWorld(true)` only on top-level object, it's recursively updates all children, and the `true` is to `updateMatrix()` inside

Reply:
You can check the source https://github.com/mrdoob/three.js/blob/dev/src/core/Object3D.js#L561